template.yaml

```diff
@@ -440,20 +484,43 @@ Resources:
       AlternateDomainNames: !Ref AlternateDomainNames
       OAuthScopes: !Ref OAuthScopes
 
+  UserPoolClientUpdateHandlerServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+        Version: "2012-10-17"
+      ManagedPolicyArns:
+        - Fn::Sub: 'arn:${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+      PermissionsBoundary: !Ref PermissionBoundaryPolicyArn
+  UserPoolClientUpdateHandlerPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - cognito-idp:UpdateUserPoolClient
+              - cognito-idp:DescribeUserPoolClient
+            Resource: "*"  # * to be able to clean up after myself, if you change the User Pool Client ID
```

Comment:
can we use the user Pool arn so that we have access to exactly what we need

Reply:
@BIPLAVGHOSAL the only point of this PR is to allow https://github.com/aws-samples/cloudfront-authorization-at-edge (a public repo) to work with our boundary permission process - code you commented above belongs to original repo so any change to it is not part of this work.